controllers/autohealconfig_controller.go

```diff
@@ -47,16 +66,415 @@ type AutohealConfigReconciler struct {
+func (r *AutohealConfigReconciler) getScenarios(files *v1alpha1.AutohealConfigTree, repo *v1alpha1.ConfigRepositoryInfo, autoheal *v1alpha1.AutohealConfig, ctx context.Context) ([]v1alpha1.AutohealConfigScenario, error) {
+	log.Info(" Autoheal-operator, Stage: Retrieving Scenarios from config repo")
+
+	scenario := v1alpha1.AutohealConfigScenario{}
+	scenarios := []v1alpha1.AutohealConfigScenario{}
+
+	for _, endpoints := range files.Tree {
+		client := http.Client{}
+		repoUrl, err := url.ParseRequestURI(fmt.Sprintf("%s/contents/%s", repo.RepositoryURL, endpoints.Filename))
+		if err != nil {
+			return nil, err
+		}
+		if repo.AccessToken == "" {
+			log.Info("Config repo access token is missing, Please add to the config repo secret")
+			return nil, nil
+		}
+
+		req, err := http.NewRequest(http.MethodGet, repoUrl.String(), nil)
+		if err != nil {
+			log.Errorf("Unable to create new http request")
+			return nil, err
+
+		}
+		if repo.Tag != "" {
+			q := req.URL.Query()
+			q.Add("ref", repo.Tag)
+			req.URL.RawQuery = q.Encode()
+		}
+		req.Header.Set("Authorization", fmt.Sprintf("token %s", repo.AccessToken))
+		req.Header.Set("Accept", "application/vnd.github.v3.raw")
+
+		resp, err := client.Do(req)
+		if err != nil {
+			log.Errorf("Unable to send http request")
+			return nil, err
+
+		}
+		if resp.StatusCode != http.StatusOK {
+			log.Errorf("Non 200 response back, got status code %d ", resp.StatusCode)
+			return nil, err
+		}
+
+		defer resp.Body.Close()
+
+		bytes, err := ioutil.ReadAll(resp.Body)
+		if err != nil {
+			log.Errorf("Unable to read response body %s", err)
+			return nil, err
+
+		}
+
+		err = yaml.Unmarshal(bytes, &scenario)
+		scenarios = append(scenarios, scenario)
+		if err != nil {
+			log.Errorf("Unable to parse yaml scenarios from config repo %s", err)
+			return nil, err
+
+		}
+
+	}
+	return scenarios, nil
+}
+
+// Creates the config maps based on the scenarios gotten from github
+func (r *AutohealConfigReconciler) createConfigMaps(autoheal *v1alpha1.AutohealConfig, scenario []v1alpha1.AutohealConfigScenario, ctx context.Context) error {
+	log.Info(" Autoheal-operator, Stage: Creating  any new scenario config maps")
+
+	for _, scenarios := range scenario {
+		configmap := core.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      scenarios.Name,
+				Namespace: autoheal.Namespace,
+			},
+		}
+		query := scenarios.Query
+		label := scenarios.Label
+		script := scenarios.Script
+		graceperiod := scenarios.GracePeriod
+
+		_, err := controllerutil.CreateOrUpdate(ctx, r.Client, &configmap, func() error {
+			if configmap.Data == nil {
+				configmap.Data = make(map[string]string)
+			}
+			configmap.Labels = map[string]string{
+				"app": "autoheal",
+			}
+			configmap.Data["query"] = query
+			configmap.Data["label"] = label
+			configmap.Data["script"] = script
+			configmap.Data["gracePeriod"] = graceperiod
```

Comment:
```suggestion
                        configmap.Data["query"] = scenarios.Query
			configmap.Data["label"] = scenarios.Label
			configmap.Data["script"] = scenarios.Script
			configmap.Data["gracePeriod"] = scenarios.GracePeriod
```
And then we could remove the variable assignment above the `createOrUpdate` function.

Reply:
Yes makes total sense